packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -196,10 +202,56 @@ export default class CallDiagnosticMetrics {
   }
 
   /**
-   * TODO: NOT IMPLEMENTED
+   * Return Client Event payload by client error code
+   * @param clientErrorCode
+   * @returns
+   */
+  public getErrorPayloadForClientErrorCode(clientErrorCode: number): ClientEventError {
+    let error: ClientEventError;
+
+    if (clientErrorCode) {
+      const partialParsedError = CLIENT_ERROR_CODE_TO_ERROR_PAYLOAD[clientErrorCode];
+
+      if (partialParsedError) {
+        error = merge(
+          {fatal: true, shownToUser: false, name: 'other', category: 'other'}, // default values
+          partialParsedError
+        );
+
+        return error;
+      }
+    }
+
+    return undefined;
```

Comment:
from debug point of view it would be nice to have a fallback "Unknown" error.

Reply:
See below answer.